src/SQLProvider/Utils.fs

```diff
@@ -55,17 +55,19 @@ module internal Utilities =
         | [] -> async { () }
 
 
-    let ensureTransaction() =
+    let ensureTransaction (transactionOptions : Transactions.TransactionOptions) =
         // Todo: Take TransactionScopeAsyncFlowOption into use when implemented in Mono.
         // Without it, transactions are not thread-safe over threads e.g. using async can be dangerous)
         // However, default option for TransactionScopeOption is Required, so you can create top level transaction
         // and this Mono-transaction will have its properties.
+        let transactionScopeOption = Transactions.TransactionScopeOption.Required
+
         let isMono = Type.GetType ("Mono.Runtime") <> null
         match isMono with
-        | true -> new Transactions.TransactionScope()
+        | true -> new Transactions.TransactionScope(transactionScopeOption, transactionOptions)
```

Comment:
Ah, my bad, I think I was looking at an old version of the file for some reason!